lib/yard/templates/helpers/markup_helper.rb

```diff
@@ -30,6 +30,7 @@ def clear_markup_cache
           {:lib => :maruku, :const => 'Maruku'},
           {:lib => :'rpeg-markdown', :const => 'PEGMarkdown'},
           {:lib => :rdoc, :const => 'YARD::Templates::Helpers::Markup::RDocMarkdown'},
+          {:lib => :commonmarker, :const => 'Commonmarker'},
```

Comment:
This is problematic, as the `lib` value is a unique key passed via `-m`. This will effectively mean that the `CommonMarker` (pre 1.0?) class could never be activated. We should definitely maintain support for the old format, which would require this commonmarker key be changed-- it's not idela, but maybe `:commonmarker1`?

Reply:
`CommonMarker` seems still accessible.
https://github.com/lsegal/yard/blob/ad2c1c4d7c53f71649440c816104ef0077226cb2/lib/yard/templates/helpers/markup_helper.rb#L91-L110

---

Even if it’s not, v1.0 `CommonMaker` should have priority over pre-1.0 `Commonmarker`.
Separating a `commonmarker1` key keeps `commonmarker` out of date.
YARD does not have a system that checks provider versions and has long[^t] exposed users to Commonmarker 1.0’s backward incompatibility. We should improve this compatibility rather than make users work an extra step.

[^t]: [Commonmarker 1.0 released](https://github.com/gjtorikian/commonmarker/releases/tag/v1.0.0) last Christmas Eve!

Reply:
> Even if it’s not, v1.0 CommonMaker should have priority over pre-1.0 Commonmarker.

Unfortunately this would be a breaking change and would need a major version bump-- we're very unlikely to major version bump YARD just for a single (optional) markup library.

Reply:
YARD is currently still in 0.x versions, so I’m not sure what you meant by ‘major’.

That said, together with https://github.com/lsegal/yard/pull/1540#discussion_r1731826662, I see your concern.
YARD currently has designs tailored for CommonMarker v0.x but it’s not documented anywhere.
We can only treat Commonmarker v1.x as a separate provider.

Reply:
> YARD does not have a system that checks provider versions

We might wanna add one. But for now, …

> YARD currently has designs tailored for CommonMarker v0.x but it’s not documented anywhere.

… I’ll add a comment in the code since it’s now linked in the README.
* Prerequisite: #1544

Reply:
> At first glance it seems like YARD is no longer able to support the pre 1.0 commonmarker lib due to a change in the markup_helper lookup table.

Actually, we might not be able to support both Commonmarker and CommomMarker at all.
Commonmarker 1.x shadows CommonMarker 0.x. We’d have to write individual `Bundler.require`s in our specs.